src/tags/cycle.ts

```diff
@@ -38,4 +39,14 @@ export default class extends Tag {
     groups[fingerprint] = idx
     return yield evalToken(candidate, ctx)
   }
+
+  public arguments (): Arguments {
+    const args = this.candidates.slice()
+
+    if (this.group) {
+      args.push(this.group)
+    }
+
+    return args
+  }
```

Comment:
Would you consider return a iterator? Not sure how do you want to comsume the results. Can be less code but not as straight forward. Just an idea, your call.

```suggestion
  public *arguments (): IterableIterator<Argument> {
    yield *this.candidates.slice()
    if (this.group) yield this.group
  }
```

Reply:
That's a good idea. In my last commit I've tried accepting an `Iterable` from `children`, `arguments`, `blockScope` and `localScope`. Now tag implementations can choose to implement them as a generator or return an array, for example, whatever is more convenient.